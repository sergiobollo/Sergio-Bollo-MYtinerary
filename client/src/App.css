.App {
    text-align: center;
    padding: 1%;
}

.App-logo {
    height: 25vmin;
    display: block;
    margin: auto;
}

.App-home {
    height: 15vmin;
    padding: 1%;
}

.App-browser {
    height: 20vmin;
    display: block;
    margin: auto;
}

.App-header {
    background-color: lightskyblue;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
}

.App-link {
    color: blue;
    padding: 5%;
}