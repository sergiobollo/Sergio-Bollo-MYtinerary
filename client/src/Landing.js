import React from 'react';

function Logo(props) {
    return <img src = { props.source }
    alt = { props.alt }
    className = "App-logo" /
        >
    ;
}

function Browser(props) {
    return <a href = { props.href } > < img src = { props.src }
    alt = { props.alt }
    className = "App-browser" /
        >
        <
        /a >
}

function FindTrip(props) {
    return <
        div > < p > Find your perfect trip, designed by < br / >
        insiders who know and love their cities. < /p> <
    h1 > Start Browsing < /h1> <
    Browser href = { props.link }
    src = { props.source }
    alt = { props.name }
    / > < /
    div >
}

function LogIn(props) {
    return <a href = { props.href }
    className = "App-link" > Log in < /a>
}

function CreateNewAccount(props) {
    return <a href = { props.href }
    className = "App-link" > Create Account < /a>
}

function BuildMytineray(props) {
    return <div >
        <
        p > Whant to build your own MYtinerary ? < /p> <
    LogIn href = { props.linkLogin }
    / > <
    CreateNewAccount href = { props.linkCreateAccount }
    /> < /
    div >
}

function Home(props) {
    return <a href = { props.link } > < img src = { props.source }
    alt = { props.alt }
    className = "App-home mt-4" /
        >
        <
        /a >
}

function Landing() {
    return <div >
        <
        Logo source = "./MYtineraryLogo.png"
    alt = "Logo" / >

        <
        FindTrip source = "./circled-right-2.png"
    name = "Browser"
    link = "./Cities" / >

        <
        BuildMytineray linkLogin = "./Login"
    linkCreateAccount = "./CreateAccount" / >
        <
        Home link = "/"
    source = "./homeIcon.png"
    alt = "Home" / >
        <
        /div>
}

export default Landing;