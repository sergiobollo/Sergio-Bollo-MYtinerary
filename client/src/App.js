import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import React, { Component } from 'react';
import { BrowserRouter, Route, Switch, Link } from 'react-router-dom';
import Landing from './Landing';
import Cities from './Cities';
import Login from './Login';
import CreateAccount from './CreateAccount';
import Itinerary from './itinerary';
import Notfound from './notfound';

export default class App extends Component {
    render() {
        return ( < BrowserRouter basename = '/' >
            <
            div className = "App" >
            <
            ul >
            <
            li >
            <
            Link to = "/" > Home < /Link> < /
            li > <
            li >
            <
            Link to = "/Cities" > Cities < /Link> < /
            li > <
            li >
            <
            Link to = "/Login" > Login < /Link> < /
            li >
            <
            li >
            <
            Link to = "/CreateAccount" > Create Account < /Link> < /
            li >
                <
            li >
            <
            Link to = "/itinerary" > Itinerary < /Link> < /
            li >
                < /
            ul >
            <
            hr / >

            <
            Switch >
            <
            Route exact path = '/'
            component = { Landing }
            /> <
            Route path = '/Cities'
            component = { Cities }
            />  <
            Route path = '/Login'
            component = { Login }
            />  <
            Route path = '/CreateAccount'
            component = { CreateAccount }
            /> 
                <
            Route path = '/itinerary'
            component = { Itinerary }
            /> 
                <
            Route component = { Notfound }
            /> < /
            Switch > <
            /
            div >

            <
            /
            BrowserRouter >
        );
    }
}
