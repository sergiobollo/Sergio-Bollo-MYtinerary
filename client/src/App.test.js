import React from 'react';
import { render } from '@testing-library/react';
import toJson from 'enzyme-to-json';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Landing from './Landing';
import App from './App'

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( < App / > , div);
    ReactDOM.unmountComponentAtNode(div);
});

it('Landing component should render without crashing', () => {
    shallow( < Landing / > );
});

it("landing component snapshot with Enzyme", () => {
    const wrapper = shallow( < Landing / > );
    expect(toJson(wrapper)).toMatchSnapshot();
});