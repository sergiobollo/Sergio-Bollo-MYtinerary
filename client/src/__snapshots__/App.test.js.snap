// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing component snapshot with Enzyme 1`] = `
<div>
  <Logo
    alt="Logo"
    source="./MYtineraryLogo.png"
  />
  <FindTrip
    link="./Cities"
    name="Browser"
    source="./circled-right-2.png"
  />
  <BuildMytineray
    linkCreateAccount="./CreateAccount"
    linkLogin="./Login"
  />
  <Home
    alt="Home"
    link="/"
    source="./homeIcon.png"
  />
</div>
`;
